vyper/semantics/validation/utils.py

```diff
@@ -295,6 +293,29 @@ def types_from_UnaryOp(self, node):
         return _validate_op(node, types_list, "validate_numeric_op")
 
 
+def _is_empty_list(node):
+    """
+    Checks if a node is a `List` node with an empty list for `elements`,
+    including any nested `List` nodes.
+
+    Arguments
+    ---------
+    node: vy_ast.VyperNode
+        A Vyper node
+
+    Returns
+    -------
+    bool
+        Boolean value indicating if the node is an empty `List` node.
+    """
+    if not isinstance(node, vy_ast.List):
+        return False
+
+    if any(isinstance(i, vy_ast.List) for i in node.elements):
+        return any(_is_empty_list(i) for i in node.elements)
+    return all(isinstance(i, vy_ast.List) and not i.elements for i in node.elements)
```

Comment:
nit: this condition was kind of confusing to me, especially the check on line 315. i think it will be clearer if the conditions are inverted, and the recursive case comes either first or last, something like:
```suggestion
    if not node.elements:
        return True
    return all(_is_empty_list(t) for t in node.elements)
```